File: dotnet-desktop-guide/net/winforms/controls-design/snippets/how-to-create-usercontrol/vb/ClearableTextBox.vb
'<default_event>
Imports System.ComponentModel

<DefaultEvent("TextChanged")>
Public Class ClearableTextBox
    '</default_event>
    '<text_changed>
    <Browsable(True)>
    Public Shadows Custom Event TextChanged As EventHandler
        AddHandler(value As EventHandler)
            AddHandler txtValue.TextChanged, value
        End AddHandler
        RemoveHandler(value As EventHandler)
            RemoveHandler txtValue.TextChanged, value
        End RemoveHandler
        RaiseEvent(sender As Object, e As EventArgs)

        End RaiseEvent
    End Event
    '</text_changed>

    '<text>
    <Browsable(True)>
    Public Shadows Property Text() As String
        Get
            Return txtValue.Text
        End Get
        Set(value As String)
            txtValue.Text = value
        End Set
    End Property
    '</text>

    '<title>
    <Browsable(True)>
    Public Property Title() As String
        Get
            Return lblTitle.Text
        End Get
        Set(value As String)
            lblTitle.Text = value
        End Set
    End Property
    '</title>

    '<click_event>
    Private Sub btnClear_Click(sender As Object, e As EventArgs)
        txtValue.Text = ""
    End Sub
    '</click_event>
End Class

File: dotnet-desktop-guide/net/winforms/controls-design/snippets/how-to-create-usercontrol/csharp/ClearableTextBox.cs
//<default_event>
using System.ComponentModel;

namespace UserControlProject
{
    [DefaultEvent(nameof(TextChanged))]
    public partial class ClearableTextBox : UserControl
    //</default_event>
    {
        //<text_changed>
        [Browsable(true)]
        public new event EventHandler? TextChanged
        {
            add => txtValue.TextChanged += value;
            remove => txtValue.TextChanged -= value;
        }
        //</text_changed>

        //<text>
        [Browsable(true)]
        public new string Text
        {
            get => txtValue.Text;
            set => txtValue.Text = value;
        }
        //</text>

        //<title>
        [Browsable(true)]
        public string Title
        {
            get => lblTitle.Text;
            set => lblTitle.Text = value;
        }
        //</title>

        public ClearableTextBox() =>
            InitializeComponent();

        //<click_event>
        private void btnClear_Click(object sender, EventArgs e) =>
            Text = "";
        //</click_event>
    }
}
